config/region.go

```diff
@@ -93,18 +98,20 @@ func EnsureRegionsCache(ctx context.Context, c Config, opts GetRegionOpts) (Regi
 	regionsCache.ExpireAt = time.Now().Add(10 * time.Minute)
 
 	if opts.SkipAuth {
+		debug.Println("[Regions] Do not save the cache")
 		// If we skipped the authentication the region cache should not be saved since it will not contain regions that are not publicly available (like osc-secnum-fr1)
 		return regionsCache, nil
 	}
 
+	debug.Println("[Regions] Save the cache")
 	fd, err = os.OpenFile(c.RegionsCachePath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0750)
-	if err == nil {
-		json.NewEncoder(fd).Encode(regionsCache)
-		fd.Close()
-	} else {
-		debug.Printf("[Regions] Failed to save the regions cache: %v\n", err)
+	if err != nil {
+		debug.Printf("[Regions] Failed to save the cache: %v\n", err)
+		return regionsCache, nil
 	}
```

Comment:
We are more used to check `err != nil` in Go. So I'm refactoring to use this pattern here.